server/src/main/java/org/elasticsearch/index/mapper/DocumentParserContext.java

```diff
@@ -279,6 +279,9 @@ final ObjectMapper getDynamicObjectMapper(String name) {
      * Add a new runtime field dynamically created while parsing.
      */
     public final void addDynamicRuntimeField(RuntimeField runtimeField) {
```

Comment:
I looked into enforcing that this is called only for new runtime fields. That is true right now but we don't enforce it anywhere. Enforcing it though looks overkill, as it adds potential overhead and I am not even sure how worthwhile it is. Should we start by making this method package protected?

Reply:
addressed in dbf2022